fuse_core/include/fuse_core/publisher.hpp

```diff
@@ -82,7 +82,8 @@ class Publisher
    *
    * @param[in] name A unique name to give this plugin instance
    */
-  virtual void initialize(const std::string & name) = 0;
+  template<typename NodeT>
+  void initialize(NodeT interfaces, const std::string & name, size_t thread_count = 1);
```

Comment:
I was having issues with this one; The problem comes from the fact that the publishers are supposed to be plugins that are loaded in by pluginlib. So all derived classes **must** have a constructor with no arguments, so there'd be no way to configure it using constructor args on the derived class (which means the thread count will have to be defined in the class declaration statically (or otherwise as a template argument), or initialized later on (which is the option I went with.)

Let me try removing the arg on `publisher.hpp` and deferring it to AsyncPublisher instead, maybe that'll work? :thinking:

Reply:
https://github.com/locusrobotics/fuse/compare/f9d587a8c481bab1398f71dd2a4a5eb28fa82547..6ee89e8d6b24b282fb99c14482fe2d6412dcc3e9

Reply:
>  so there'd be no way to configure it using constructor args on the derived class (which means the thread count will have to be defined in the class declaration statically (or otherwise as a template argument), or initialized later on (which is the option I went with.)

True. It looks like the ROS 1 code we're porting went with defining the value in the subclass statically. I opened a PR on your fork that restores that for `AsyncPublisher` https://github.com/methylDragon/fuse/pull/3

Initializing it later on is a little problematic because `pluginlib` requires all derived classes to have the same signature for `initialize()` https://github.com/locusrobotics/fuse/pull/299#discussion_r1058533703 - that means we'd have to make all publishers accept a `thread_count` argument even if they didn't derive from `AsyncPublisher`.